src/integration/integration_api_broker_graceful_shutdown_test.go

```diff
@@ -0,0 +1,177 @@
+package integration_test
+
+import (
+	"fmt"
+	. "integration"
+
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+)
+
+var _ = Describe("Integration_Api_Broker_Graceful_Shutdown", func() {
+
+	const (
+		MessageForServer = "message_for_server"
+	)
+
+	BeforeEach(func() {
+		fakeScheduler = ghttp.NewServer()
+		fakeApiServer := ghttp.NewServer()
+
+		apiServerConfPath = components.PrepareApiServerConfig(components.Ports[APIServer], dbUrl, fakeScheduler.URL(), tmpDir)
+		serviceBrokerConfPath = components.PrepareServiceBrokerConfig(components.Ports[ServiceBroker], brokerUserName, brokerPassword, dbUrl, fakeApiServer.URL(), brokerApiHttpRequestTimeout, tmpDir)
+
+	})
+
+	AfterEach(func() {
+		stopAll()
+	})
+
+	Describe("Graceful Shutdown", func() {
+
+		Context("ApiServer", func() {
+
+			BeforeEach(func() {
+				startApiServer()
+			})
+
+			It("stops receiving new connections after being interrupted", func() {
```

Comment:
Yes. Isn't that what we expect?
You have hidden some of that check in the sendsignal function which is wrong.
In theory, the sigusr2 should stop accepting connections and when the last connection is closed have it shut down.
So it should be open a persistent connection, sendsigusr2, no new connections allowed, process is running, close connection, process has exited.